### PR TITLE
New release mechanism

### PR DESCRIPTION
**Pull request by THenry14**
## Describe your changes

This PR changes the flow of package release to maven. 
In order to release bump a version in gradle build file and create a new tag matching this version. This will trigger build an publish flow.

## Linked issues

Closes https://github.com/software-mansion/starknet-jvm/issues/68

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,2 +1,3 @@
 android.useAndroidX=true
-org.gradle.jvmargs=-XX:MaxMetaspaceSize=512m
\ No newline at end of file
+org.gradle.jvmargs=-XX:MaxMetaspaceSize=512m
+gnsp.disableApplyOnlyOnRootProjectEnforcement=true
```
Comment (cptartur) [inline gradle.properties:3]: Why was this added?

File: lib/build.gradle.kts [hunk 2]
```diff
@@ -141,70 +142,69 @@ publishing {
...
+                    val okHttpDependency = dependencyNode.appendNode("dependency")
+                    okHttpDependency.appendNode("groupId", "com.squareup.okhttp3")
+                    okHttpDependency.appendNode("artifactId", "okhttp")
+                    okHttpDependency.appendNode("version", "4.10.0")
+
+                    val kotlinxDependency = dependencyNode.appendNode("dependency")
+                    kotlinxDependency.appendNode("groupId", "org.jetbrains.kotlinx")
+                    kotlinxDependency.appendNode("artifactId", "kotlinx-serialization-json")
+                    kotlinxDependency.appendNode("version", "1.3.3")
                 }
```
Comment (cptartur) [inline lib/build.gradle.kts:204]: How complicated it'd be to include these automatically based on main build.gradle.kts? That way we won't have to remember to update these every time we bump a dependency version.

Comment (THenry14) [reply lib/build.gradle.kts:204]: The problem is, we have mutliple deps decalred in the main build file, and not all of them should be noted as dependencies for the package (eg. test stuff). So, not sure how complex, but I imagine quite, as we still would have to somehow specify what goes in here one way or the other

File: gradle.properties (hunk 1 above)
Comment (THenry14) [reply gradle.properties:3]: >* What went wrong:
An exception occurred applying plugin request [id: 'io.codearte.nexus-staging']
> Failed to apply plugin 'io.codearte.nexus-staging'.
   > Nexus staging plugin should ONLY be applied on the ROOT project in a build. See https://github.com/Codearte/gradle-nexus-staging-plugin/issues/47 for explanation. If you really know what you are doing it can be overridden with setting 'gnsp.disableApplyOnlyOnRootProjectEnforcement' property.
   
The problem is, we have multi-project setup, and I cannot take the whole lib' `publishing` block to the root

File: lib/build.gradle.kts (hunk 2 above)
Comment (cptartur) [reply lib/build.gradle.kts:204]: I see we already used hardcoded dependencies in the previous version. Could you open an issue for making these automatically included? For the time being, we can merge it as is.

Comment (THenry14) [reply lib/build.gradle.kts:204]: https://github.com/software-mansion/starknet-jvm/issues/143

**Review by cptartur — APPROVED** [review]
Perhaps we should add a information to README.md or in other place that creating a tag is needed for a new version release?
